src/SFML/Graphics/Shape.cpp

```diff
@@ -143,6 +145,38 @@ FloatRect Shape::getGlobalBounds() const
     return getTransform().transformRect(getLocalBounds());
 }
 
+Vector2f Shape::getGeometricCenter() const
+{
+    const auto count = getPointCount();
+
+    switch (count)
+    {
+        case 0:
+            err() << "Shape::getGeometricCenter() called on empty Shape";
+            std::terminate();
```

Comment:
Is it better to `assert(false)` and return a default-constructed `Vector2f`? I'm still on the fence about termination as opposed to returning some value. The goal is to avoid UB and returning `(0, 0)` constitutes defined behavior in release builds. This is the only place we call `std::terminate` within SFML so it stands out in that regard.

Reply:
I'm not sure what's best. I had a default constructed `Vector2f` as the return value before but was adviced to change it to `std::terminate` in a CR. `{0, 0}` wouldn't be _too_ crazy if there are no points, at least not in this context I think.

Reply:
Haha yeah sorry if we've taken you in circles. Don't go changing it yet until we can have a final discussion about it.

Reply:
No worries! :-) I'll leave it as-is for now.

Reply:
I would also not use `std::terminate`, as you say it'd be the only place where this happens.

`assert` + returning a zero-vector is probably sane. https://github.com/SFML/SFML/issues/2139 is unfortunately not concluded yet...

Reply:
Ok, I've changed back to an `assert` and returning `Vector2f{}` in my latest patch :)